runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -2154,6 +2154,9 @@ class INTERPRETER_CLASS
 		 * After the synchronization (if any) is done, the visible bit will be or'ed in.
 		 */
 		bp = buildSpecialStackFrame(REGISTER_ARGS, J9SF_FRAME_TYPE_JNI_NATIVE_METHOD, jitStackFrameFlags(REGISTER_ARGS, 0), true);
+#if JAVA_SPEC_VERSION >= 19
```

Comment:
There are failure cases which will miss the decrement. I suggest putting the inc/dec immediately before and after the call to `callCFunction`.